zend_abstract_interface/zai_string/string.h

```diff
@@ -5,28 +5,82 @@
 #include <stddef.h>
 #include <string.h>
 
+#include <Zend/zend.h>
+
+/**
+ * Represents a non-owning view of a string.
```

Comment:
`zai_symbol_lookup_key` at least returns an owned string. ... And it has to case the `const` away on ptr to free it :-)
There likely should be `zai_string_owned` with a non-const ptr. And a possibility to alias the owned struct to the view.

Reply:
E.g.
```
typedef union zai_string_owned_s {
    struct {
        size_t len;
        char *ptr;
    };
    zai_string_view view;
} zai_string_owned;
```